pkg/proxy/record.go

```diff
@@ -93,18 +104,13 @@ type LogRecord struct {
 	ObservationPoint ObservationPoint
 
 	// SourceEndpoint is information about the soure endpoint if available
-	SourceEndpoint *EndpointInfo `json:"SourceEndpoint,omitempty"`
+	SourceEndpoint EndpointInfo
 
 	// DestinationEndpoint is information about the soure endpoint if available
-	DestinationEndpoint *EndpointInfo `json:"DestinationEndpoint,omitempty"`
-
-	// Source is the IP and port of the endpoint that generated the
-	// request
-	Source IPPort
+	DestinationEndpoint EndpointInfo
 
-	// Destination is the IP and port of the endpoint that is receiving the
-	// request
-	Destination IPPort
+	// IPVersion indicates the version of the IP protocol in use
+	IPVersion IPVersion
```

Comment:
Actually, the IPv4 and IPv6 is defined as a string. So, the proxy can log "" if, as an example, the node doesn't have an IPv6 address. So, I think we're fine. We don't need an additional field to tell us which field is valid (v4 or v6 address).